### ISSUE TITLE
Configuration file: [demo2] key

### ISSUE THREAD

**Issue by mturilli** [radical-cybertools/radical.owms#11]
- Can it be changed? Is it arbitrary? 

More in general:
- Is there a schema/protocol for the configuration files/keys?
- [round_robin_overlay_scheduler]: Can we have multiple 'resources' keys?
- [general]: to what it applies?
- Multiple semantic components in the same configuration file makes difficult to understand it. 

Extensibility:
- How do we add new sections/keys for plugins to the conf(s) file?

**Comment by andre-merzky** [comment]
The `phase1_demo_2.py` contains this piece of code:

```
    session = troy.Session ()
    config  = session.cfg.get_config_as_dict ()

    if 'demo2' in config : demo_cfg = config['demo2']
    else                 : demo_cfg = dict ()
```

This retrieves the global troy config from the `troy.Session` instance, and then digs out the 'demo2' section, if that exists.  So, that part of the config file is very specific to this demo script - hence the specific section name.  But we use a similar mechanism to define similar sections for, say, a specific plugin:

```
[default_overlay_translator]
pilot_size = 4
```

is made available in the default  plugin's `init()` call,  via:

```
self.cfg = cfg.as_dict ().get (self.name, {})
```

`self.name' is derived from the plugin description, so the above translates to 

```
self.cfg = cfg.as_dict ().get ('overlay_translator_default', {})
```

The `dict.get (name, default)` syntax makes sure to return an empty dict if the troy config does not define that section.

Now, that code above is right now explicit, both in the demo script and in the plugins.  For the plugins, it would be easy to make this implicit on plugin loading, and to automatically equip each plugin with a self.cfg -- but that would be (a) limiting, and (b) would imply a somewhat stricter definition on plugin structure -- which we tried to avoid so far (see discussion thread on the planner plugin).

About (a), limiting: the 'bundle' ini section for example may be needed by multiple plugins -- so we don't want to limit the plugin's config access to one specific subsection -- that is why we pass the complete troy config to the plugin's init call, and let the plugin dig out any sections it might need -- see planner plugin which several ini sections...

Hmm, this is probably more detail than you wanted -- and I am still not sure I answered your actual question... :P  ;)
